File: Python/Misc_ProblemSolving/Two Sum.py
#Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

#leetcode : https://leetcode.com/problems/two-sum/

#Python Code :

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        dic={}
        for i in range(n):
            if target-nums[i] in dic:
                return [dic[target-nums[i]],i]
            else:
                dic[nums[i]]=i

File: 1-two-sum/two-sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i =0; i < nums.length; i++){
            int remaining = target - nums[i];
            if(map.containsKey(remaining)){
                return new int[] {map.get(remaining), i};
            }
            map.put(nums[i], i);
        }
        return new int[] {-1, -1};
    }
}